objects/actors/mob/Mob.gd

```diff
@@ -1,4 +1,24 @@
 class_name Mob
 extends Actor
 
+@onready var healthbar = $HealthBar
+@onready var animationPlayer = %AnimationPlayer
+
+@export var maxHealth: int = 10
+@export var currentHealth: int = maxHealth
+
 var targeted_players: Array[Player]
+
+func _ready():
+	healthbar.init(currentHealth)
+
+func apply_attack(damage: int):	
+	if currentHealth <= 0: return
+	
+	currentHealth -= damage
+	healthbar.value = currentHealth
+	animationPlayer.play("Hit")
+	
+	if currentHealth <= 0:
+		animationPlayer.play("Death")
+		set_physics_process(false)
```

Comment:
We may have this code within `Actor` like this:
```gdscript
func apply_attack(damage: int):
	if currentHealth <= 0: return
	
	currentHealth -= damage
	healthbar.value = currentHealth
	_on_hit()
	
	if currentHealth <= 0:
		_on_death()

func _on_hit(): pass
func _on_death(): pass
```

And `_on_hit()` & `_on_death()` will be overriden in `Mob` or `Player`.